### PR TITLE
Added typings

### PR DESCRIPTION
**Pull request by SachsKaylee**
Merging the types into the main repository is definitely better than having a separate one, I agree(See PR #24).

Let me just explain the added files:

- **d/index.d.ts** - The definitions live here.
- **tsconfig.json** - Includes directives for the project that includes sockette on where to find the types, etc. Not a lot since we don't actually compile sockette using the TS compiler.

And obviously an added hint in the Readme. 
I just specified the latest Version of TypeScript in the package.json, although it should probably work with older ones as well. We're not really using any special features.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -160,6 +160,9 @@ If [`options.maxAttempts`](#optionsmaxattempts) has not been exceeded, enqueues
 
 Initializes a new `WebSocket` &mdash; used on initialization and by [`reconnect()`](#reconnect).
 
+## TypeScript
```
Comment (lukeed) [inline readme.md]: Let's not include this. Those who have TS on will see it automatically.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (lukeed) [inline tsconfig.json]: _Pretty sure_ we don't need this file either.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "sockette",
   "version": "2.0.0",
+  "typeScriptVersion": "2.8",
```
Comment (lukeed) [inline package.json]: Don't need this bad boy.

**Review by lukeed — CHANGES_REQUESTED** [review]
Hey, thanks for this!

I think this may be more than what we actually need for including typings.

A few of my repos have been PR'd with types & all you really need is the `*.d.ts` to be added, add that same file to `files` array in `package.json` and then add a new `"types": "{NAME}.d.ts"` to the `package.json` as well.

If things have drastically changed, let me know, but I dont imagine that they have.

**Comment by SachsKaylee** [comment]
Okay, thanks for the info! Is the entry in files in package.json required? In my test environment it works without it. I also couldn't find any reference to this in the official guide: [https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)

**Comment by lukeed** [comment]
Yeah, it's required. The `npm` registry only saves/uploads the contents of "files"  when specified. So it works on your machine because it exists for you already.

**Review by lukeed — APPROVED** [review]

**Comment by lukeed** [comment]
Looks good, thank you~!

**Pull request merged**